### ISSUE TITLE
Unwanted folder duplication within base folder

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#11203]
A user contacted us with the problem, that his base folder `/cgeo/` contained hundreds of the subfolders for GPX and maps.
One named `gpx` and the others like `gpx (xxx)` counting up; same for maps. All duplicates are empty.

First I could not think of any reasonable scenario where this could happen, but finally I stumbled over the fact, that especially the users GPX and maps folders contain capital letters in their definition (see sys info below). They are named `GPX` and `Maps`.

Is it possible, that at the point in our code where we check for existence of these folders (maybe in the legacy non-SAF part) we somehow loose the capitalization and end of with trying to create whats already existing?

The user could not tell, whether it started with the current beta or already on the last release.

```
c:geo version: 2021.07.10-RC

Device:
-------
- Device type: MI 8 Pro (equuleus, Xiaomi)
- Available processors: 8
- Android version: 10
- Android build: QKQ1.190828.002 test-keys
- Screen resolution: 1080x2029px (392x737dp)
- Sailfish OS detected: false
- Google Play services: enabled - 21.24.18 (120400-383468479)
- HW acceleration: enabled (default state)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Hide caches: own/ foundarchived
- Hide waypoints: -
- Set language: en_GB (system default)
- System date format: dd/MM/y
- Debug mode active: no
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Global filter: display all caches
- Last backup: never
- Routing mode: Car
- Settings: v5, Count:151
- Map: CacheMapUK_whole_UK_2016 (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:cgeo/Maps/CacheMapUK_whole_UK_2016.map
  - Atts: mapsforge-map-writer-0.5.1
  - Theme: none

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
   Geocaching.com Adventure Lab
- Geocaching.com date format: dd/MM/yyyy
- Routing: external / BRouter installed: false
- Installed c:geo plugins:  none

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (88.3 GB free) v2 internal isDir(9 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (88.3 GB free) v2 external non-removable isDir(33 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (88.3 GB free) v2 external non-removable isDir(4829 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (88.3 GB free) v2 internal isDir(18 entries)
- Public Folders: #11
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=8, dirs:>=34, totalFileSize:>=228.3 KB, free space: 88.3 GB, files on device: 7413760)
  - OFFLINE_MAPS: /cgeo/Maps (User-Defined)[/cgeo/Maps[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FMaps::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FMaps/document/primary%3Acgeo%2FMaps, Av:true, files:21, dirs:3, totalFileSize:3.1 GB, free space: 88.3 GB, files on device: 7413760)
  - OFFLINE_MAP_THEMES: /cgeo/Maps/Themes/CacheMapUK_icons (User-Defined)[/cgeo/Maps/Themes/CacheMapUK_icons[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FMaps%2FThemes%2FCacheMapUK_icons::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FMaps%2FThemes%2FCacheMapUK_icons/document/primary%3Acgeo%2FMaps%2FThemes%2FCacheMapUK_icons, Av:true, files:17, dirs:0, totalFileSize:20.4 KB, free space: 88.3 GB, files on device: 7413760)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:9, dirs:0, totalFileSize:1.3 MB, free space: 88.3 GB, files on device: 7413760)
  - GPX: /cgeo/GPX (User-Defined)[/cgeo/GPX[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FGPX::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FGPX/document/primary%3Acgeo%2FGPX, Av:true, files:1, dirs:0, totalFileSize:147.2 KB, free space: 88.3 GB, files on device: 7413760)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 88.3 GB, files on device: 7413760)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 88.3 GB, files on device: 7413760)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 88.3 GB, files on device: 7413760)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81.2 KB, free space: 88.3 GB, files on device: 7413760)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 88.3 GB, files on device: 7413760)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 88.3 GB, files on device: -1)
- Map render theme path:
- PersistedDocumentUris: #1
- TRACK: null
- Persisted Uri Permissions: #4
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (29 Mar, 07:44):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FGPX (29 Mar, 07:44):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FMaps%2FThemes%2FCacheMapUK_icons (29 Mar, 07:44):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo%2FMaps (29 Mar, 07:44):RW
- Database: /storage/emulated/0/Android/data/cgeo.geocaching/files/databases/data (v95, Size:98.3 MB) on user storage

--- End of system information ---

**Comment by Lineflyer** [comment]
Questions handed over to user, waiting for feedback.

Ticket: 564004

**Comment by eddiemuc** [comment]
I played around with this in emulator and had some weird findings:
* I created an uppercase folder name (GPXNEW). When I renamed this folder to lowercase (gpxnew) using my "File" app on the emulator (not c:geo), I ended up with a folder named "gpxnew (1)". Seems like not even Android itself can handle all these case correctly.
* I registered the uppercase folder with c:geo (GPXNEW), then renamed the folder on the storage to lowercase. After that using it from c:geo failed (expected). But when I re-selected the folder, it still appeared in UPPERCASE in the selection dialog and I was able to select it again. After that, although the folder is still registered in UPPERCASE (!) in c:geo, it kept writing to the LOWERCASE (real) folder on disk. No duplication.

I assume the above can be different for different underlying storage systems, but I have no real idea what to do here. I guess it is best to advise the user to re-grant access to the folders and use lowercase folder names. Unless he has a scenario where this can be cleary reproduced.

(The user in quesion should best use the default folder values btw, he grants access to same-named subfolders of cgeo anyway)

**Comment by Lineflyer** [comment]
This is the users feedback:

---

Hi, When I reported the problem the folders were /cgeo/GPX and cgeo/Maps and the new folders created with the number suffix were all lowercase.

When exporting to gpx the files all went in to the /cgeo/GPX folder and the newly created ones remained empty.

After receiving your previous message which indicated that the folder names should be lowercase, I changed them to cgeo/gpx and cgeo/maps and have had no further occurrences of extra folders being created.

---

Lets close here. Seems not to be "our" bug but something buried in the system.
